packages/jest-haste-map/src/worker.js

```diff
@@ -52,11 +52,15 @@ export async function worker(data: WorkerMessage): Promise<WorkerMetadata> {
   // Process a package.json that is returned as a PACKAGE type with its name.
   if (filePath.endsWith(PACKAGE_JSON)) {
     content = fs.readFileSync(filePath, 'utf8');
-    const fileData = JSON.parse(content);
-
-    if (fileData.name) {
-      id = fileData.name;
-      module = [filePath, H.PACKAGE];
+    try {
+      const fileData = JSON.parse(content);
+
+      if (fileData.name) {
+        id = fileData.name;
+        module = [filePath, H.PACKAGE];
+      }
+    } catch (err) {
+      throw new Error(`Cannot parse ${filePath} as JSON: ${err.message}`);
```

Comment:
this can be
```js
err.message = `Cannot parse ${filePath} as JSON: ${err.message}`;
throw err;
```

to reuse the stack trace of the original error instead of creating a new error.

(this is not needed if we change to `parse-json`)